File: Arrays and Stings/1.1 Is Unique/1.1 Is_Unique.js
/* ---bruteForce--- */
const bruteForce=(str)=>{
  const length = str.length
	for(let frwdIndex =0; frwdIndex<length; frwdIndex++){
  	for(let backIndex =frwdIndex+1; backIndex<length;backIndex++){
    		if(str[backIndex] === str[frwdIndex]){
        return false
        }
    }
  }
  return true
}

/* ---hashTable--- */

const hashTable =(str)=>{
	const arr={}
  for(let index=0;index<str.length;index++){
  	if(arr[str[index]] && arr[str[index]]===1){
    	return false
    }
    else{
    	arr[str[index]]=1
    }
  }
  
  return true
}

/* --- booleanArray --- */
const booleanArray=(str)=>{
  const length = str.length
	const arr = new Array(256).fill(0)
  for(var index=0;index<str.length;index++){
  	if(arr[str.charCodeAt(index)] === 1){
    return false
    }
    else{
    	arr[str.charCodeAt(index)] = 1
    }
  }
  return true
}


/* ---test--- */
bruteForce("a")
hashTable("aac")
booleanArray("aa")

File: Arrays and Stings/1.1 Is Unique/1.1 Is_Unique.py
# --- bruteFroce ---

def bruteForce(inputStr):
    for frwdIndex in range(len(inputStr)):
        for backIndex in range(frwdIndex+1,len(inputStr)):
            if(inputStr[frwdIndex] == inputStr[backIndex]):
                return False
    return True


# --- hashTable ---

def hashTable(inputStr):
    arr = {}
    for index in range(len(inputStr)):
        if inputStr[index] in arr:
            return False
        else:
            arr[inputStr[index]] = 1
    return True        
# --- booleanArray ---

def booleanArray(inputStr):
    arr=[0 for i in range(256)]
    for index in range(len(inputStr)):
        if arr[ord(inputStr[index])] == 1:
            return False
        else:
            arr[ord(inputStr[index])] =1
    return True

# --- usingSet ---

def usingSet(inputStr):
    toCheck = set(inputStr)
    if len(toCheck) != len(inputStr):
        return False
    return True

# --- setImplementation using array

def setImplementation(inputStr):
    toCheck=[]
    for i in inputStr:
        if i not in toCheck:
            toCheck.append(i)
    if len(toCheck) != len(inputStr):
        return False
    return True
    print(toCheck)
